ejb/src/main/java/org/wildfly/httpclient/ejb/ClientHandlers.java

```diff
@@ -60,12 +60,12 @@ private ClientHandlers() {
         // forbidden instantiation
     }
 
-    static HttpTargetContext.HttpMarshaller invokeHttpMarshaller(final Marshaller marshaller, final TransactionInfo txnInfo, final Object[] parameters, final Map<String, Object> contextData) {
-        return new InvokeHttpMarshaller(marshaller, txnInfo, parameters, contextData);
+    static HttpTargetContext.HttpMarshaller invokeHttpMarshaller(final Marshaller marshaller, final TransactionInfo txn, final Object[] objects, final Map<String, Object> map) {
```

Comment:
OK with objects and map, as they reflect their data types, but txn is a TransactionInfo structure and not a Transaction as the variable would lead one to believe. I would prefer txnInfo :-)

Reply:
Fixed.